core/domain/classifier_services.py

```diff
@@ -276,6 +278,25 @@ def get_classifier_training_job_by_id(job_id):
     return classifier_training_job
 
 
+def get_all_classifier_training_jobs():
```

Comment:
No need for this function if we fetch models page by page using cursor. So move this logic in `fetch_next_job` function.

Call `classifier_models.ClassifierTrainingJobModel.query_training_jobs()` from `fetch_next_job`. First call it with no cursor and second time call it with the cursor it returned on previous call (until you find a valid job).